tsconfig.json

```diff
@@ -2,12 +2,12 @@
 	"compilerOptions": {
 		/* Basic Options */
 		"target": "es5",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */
-		"module": "commonjs",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */
+		"module": "ESNext",                       /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */
 		// "lib": [],                             /* Specify library files to be included in the compilation. */
 		// "allowJs": true,                       /* Allow javascript files to be compiled. */
 		// "checkJs": true,                       /* Report errors in .js files. */
 		// "jsx": "preserve",                     /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */
-		// "declaration": true,                   /* Generates corresponding '.d.ts' file. */
+		"declaration": true,                      /* Generates corresponding '.d.ts' file. */
```

Comment:
By using this option, the following files are generated under the `dist` directory, and these will be included when distributing as npm package.

- base64.d.ts
- buffer.d.ts
- constants.d.ts
- index.d.ts

For example, `index.d.ts` has the following contents.

```typescript
import { DeckDefinition, DeckList } from "../types";
import { FormatType } from "./constants";
export declare function encode(deck: DeckDefinition): string;
export declare function decode(deckstring: string): DeckDefinition;
export { FormatType, DeckDefinition, DeckList };
```

https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html